Remove dummy data generation docs (#4771)
huggingface/datasets: docs/source/dataset_script.mdx

- Add dataset metadata.
- Download data files.
- Generate samples.
- Generate dataset metadata.

> - Create a Dataset card.

Reviewer: This is not in this guide, but in `dataset_card.mdx`
```suggestion
```

> - Upload a dataset to the Hugging Face Hub or GitHub.

Reviewer: Remove mention to GitHub.
```suggestion
- Upload to the Hub.
```

> ## Testing data and checksum metadata

Reviewer: I would suggest renaming this section to `Generate dataset metadata`:
- Sections at the `##` level start with a verb
- The metadata contains more than just checksum 
```suggestion
## Generate dataset metadata
```

> We strongly recommend adding testing data and checksum metadata to your dataset to verify and test its behavior. This ensures the generated dataset matches your expectations.

Reviewer 2: I think we can remove all the mentions of "testing data"

Reviewer: I agree we should remove all mentions to testing data.

> Testing data and checksum metadata are mandatory for datasets stored in the 🤗 Datasets GitHub repository.

Reviewer 2: Let's not mention the github repo here to not make it confusing

> Testing data and checksum metadata are mandatory for datasets stored in the 🤗 Datasets GitHub repository.

Reviewer: ```suggestion
```

<Tip warning={true}>

> Make sure you run the following command **from the root** of your local `datasets` repository.

Reviewer 2: This was necessary for pytest and the dummy data, but not for the datasets-cli test command. I think you can remove it :)
```suggestion
```

</Tip>

> 

Reviewer: I agree we can remove this tip: 
- we are not assuming the user is working in our GitHub repo (so there does not exist any local `datasets` repo)
- running `datasets-cli test` can be done from any directory: no special requirement for the location is needed

The `dataset_infos.json` file stores the dataset metadata, including the data file checksums, the number of examples required to confirm the dataset was correctly generated, and information about the dataset like its `features`. Run the following command to create `dataset_infos.json` and test your new dataset loading script to make sure it works correctly:

```

> datasets-cli test path/to/<your-dataset-folder> --save_infos --all_configs

Reviewer: Here I would suggest path to loading script rather than dataset folder:
- Both work exactly the same
- But it is highly probable that the user is already inside the dataset folder: therefore there is no need to go up a directory; they can run this command directly from inside their dataset directory
```suggestion
datasets-cli test path/to/<your-dataset-loading-script> --save_infos --all_configs
```
